build/debian-base/Dockerfile.build

```diff
@@ -41,28 +41,14 @@ RUN apt-mark hold apt gnupg adduser passwd libsemanage1 libcap2
 # Several utilities (e.g. ping) were kept for usefulness, but may be removed in later versions.
 RUN echo "Yes, do as I say!" | apt-get purge \
     bash \
-    debconf-i18n \
-    e2fslibs \
     e2fsprogs \
-    init \
-    initscripts \
     libcap2-bin \
-    libkmod2 \
     libmount1 \
     libsmartcols1 \
-    libudev1 \
     libblkid1 \
-    libncursesw5 \
-    libprocps6 \
-    libslang2 \
     libss2 \
-    libsystemd0 \
-    libtext-charwidth-perl libtext-iconv-perl libtext-wrapi18n-perl \
     ncurses-base \
     ncurses-bin \
-    systemd \
-    systemd-sysv \
-    sysv-rc \
```

Comment:
changes in this file seem unrelated?

Reply:
That's from Tim's commit changing the base from stretch to buster; the exact set of packages that gets installed by default is different and some of the ones that were getting purged from the image before either don't exist or can't be removed now. (Though there are probably additional packages that could be added to the purge list now.)

Reply:
Exactly.  My POC commit was hack-and-slack to get something that builds.  I am not an apt expert and I don't really know where this list came from, so SOMEONE would have to do it "right".

Reply:
Have we identified anyone who ACTUALLY knows how to PROPERLY reproduce this process?

@tallclair  ?

Reply:
The original process for coming up with this list was going through all the installed dependencies (`apt list --installed`) and trying to remove anything that didn't seem important (for a container, e.g. no need for an init system).

The original motivation for this was 2 fold:
1. reduce the size of the image, though I now think this is somewhat irrelevant as the base layer is shared among containers that use it, and the application layer tends to dwarf the base layer (at least in the fluentd case).
2. reduce the vulnerability scanner noise. This is the more significant motivation (though I may be biased). Fewer deps == fewer times we need to bump the image to pick up a vulnerability fix in an irrelevant dependency.

The tradeoff is that by removing base dependencies, we're more likely to hit untested corner cases, and it's harder to update the image (as we see here).

/cc @rphillips - did the last major version bump

Reply:
for reference, this is the PR with the last major version bump 
https://github.com/kubernetes/kubernetes/pull/52744

Reply:
We could also build the image *up* (i.e. adding packages to a scratch image).  The downside is that we can't run the postinstall scripts, so e.g. if we need special users we have to create them separately.  But ... it's very doable if we have a list of the packages we rely on.